### DISCUSSION TITLE
🙏  Topic data between clusters is out of sync in pulsar GEO mode

### DISCUSSION THREAD

**Discussion by luky116** [apache/pulsar#22341] [Q&A]
I used puslar's GEO for cluster replication and built a public zk as the configurationStore to synchronize the data between the two clusters. I found a problem. The steps are as follows, can someone do me a falvor 🙏 🙏 🙏 :

1. Suppose there are two clusters synchronizing data with each other, cluster1 and cluster2; 
2. I create a Namespace in cluster1 named ns1, ns1 Only belongs to cluster1; 
3. Create a Topic under ns1, named topic1, and produce 100 messages to Topic1; 
4. Set ns1 to belong to cluster1 and cluster2 through the set-clusters command; 
5. 【Question 1 】At this time, you will find that topic1 does not exist in cluster2. This topic1
6. 【Question 2】We use the command to create topic1 in cluster2, and I am prompted that "topic1 already exists", so the creation failed.

<img width="1480" alt="image" src="https://github.com/apache/pulsar/assets/38887641/cb9ed346-5874-4632-ac21-f919c6c5ea01">

Pulsar version：V3.0.3

**Comment by david-streamlio** [comment]
After you create the namespace (ns1) in step 2, does the namespace exist on both clusters? 

After you create the topic (ns1/topic1) in step 3, but before you publish messages, does the topic exist on both clusters?

**Reply by luky116** [reply]
1、after step 2, the namespace exist on both clusters.
2、cluster2 donnot have permission to see topic1, because ns1 belongs to cluster1
![image](https://github.com/apache/pulsar/assets/38887641/def3c139-4287-4067-86af-357abd508e68)
3、i use set-clusters command to make ns1 belongs to both cluster1 and cluster2, but cluster2 still donot have topic1
![image](https://github.com/apache/pulsar/assets/38887641/5c859273-7a9c-4c52-bd1b-b0ea81c58391)
4、I try to create topic2 in cluster2, but it returns “ Partitioned topic already exists”
![image](https://github.com/apache/pulsar/assets/38887641/f2c2b934-a179-4098-a9a9-b81b951d52ad)

**Comment by david-streamlio** [comment]
You should use the `--clusters` switch when you [create](https://pulsar.apache.org/reference/#/3.2.x/pulsar-admin/namespaces?id=create) the namespace.
